### PR TITLE
feat (api): workflow run tags

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/workflow/dao_run.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 
 	"github.com/go-gorp/gorp"
 
+	"strings"
```
Comment (yesnault) [inline engine/api/workflow/dao_run.go]: import

File: engine/api/workflow/dao_run.go [hunk 2]
```diff
@@ -47,6 +49,10 @@ func (r *Run) PostInsert(db gorp.SqlExecutor) error {
 		return sdk.WrapError(err, "Run.PostInsert> Unable to store marshalled infos")
 	}
 
+	if err := updateTags(db, r); err != nil {
+		return sdk.WrapError(err, "Run.PostInsert> Unable to store insert tags")
```
Comment (yesnault) [inline engine/api/workflow/dao_run.go]: to store update tags

File: engine/api/workflow/process.go [hunk 3]
```diff
@@ -310,6 +310,13 @@ func processWorkflowNodeRun(db gorp.SqlExecutor, w *sdk.WorkflowRun, n *sdk.Work
 	}
 	run.BuildParameters = jobParams
 
+	for _, p := range jobParams {
+		switch p.Name {
+		case "git.hash", "git.branch", "git.tag":
```
Comment (sguiheux) [inline engine/api/workflow/process.go]: git.author?

File: engine/sql/035_workflow_run_tags.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS "workflow_run_tag" (
+    workflow_run_id BIGINT,
+    tag VARCHAR(256) NOT NULL,
+    value VARCHAR(256) NOT NULL,
+    PRIMARY KEY (workflow_run_id, tag)
+);
+
+SELECT create_foreign_key('FK_WORKFLOW_RUN_TAG_WORKFLOW_RUN', 'workflow_run_tag', 'workflow_run', 'workflow_run_id', 'id');
```
Comment (sguiheux) [inline engine/sql/035_workflow_run_tags.sql]: Delete cascade

**Pull request merged**